misc/windows-deploy/hostsetup.ps1

```diff
@@ -11,21 +11,21 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-$networkAdapters = Get-CimInstance -ClassName Win32_NetworkAdapter -Filter "AdapterTypeId='0' AND NetEnabled='True' AND NOT Name LIKE '%Hyper-V Virtual%' AND NOT Name LIKE '%Loopback Adapter%' AND NOT Name LIKE '%TAP-Windows Adapter%'" | Sort-Object -Property "Index"
-if(-not $networkAdapters -or $networkAdapters.Length -eq 0)
-{
-    throw New-Object System.Exception("Failed to find the primary network interface")
-}
-$ifIndex = $networkAdapters[0].InterfaceIndex
-$networkAdapterConfig = Get-CimInstance -ClassName Win32_NetworkAdapterConfiguration -Filter "InterfaceIndex='$ifIndex'" | select IPConnectionMetric, DefaultIPGateway
-$defaultGateway = $networkAdapterConfig.DefaultIPGateway[$networkAdapterConfig.DefaultIPGateway.Length - 1]
+$ErrorActionPreference = 'Continue'
+
+$adapter = (Get-NetAdapter -Name "*APIPA*")
 
-$containerGateway = (Get-NetNat).InternalIPInterfaceAddressPrefix | %{ $_ -replace '/.*$', '' }
+if(!($adapter)) {
+	Add-VMNetworkAdapter -VMNetworkAdapterName "APIPA" -SwitchName nat -ManagementOS
+}
 
-netsh interface portproxy add v4tov4 listenaddress=$containerGateway listenport=51679 connectaddress=127.0.0.1 connectport=51679 protocol=tcp
+$ifIndex = (Get-NetAdapter -Name "*APIPA*" | Sort-Object | Select ifIndex).ifIndex
 
-$existingRoute=route print 169.254.170.2 -4 | Sort-Object -Property "IPv4 Route Table"
-if ($existingRoute -Match "169.254.170.2") {
-   route -p delete 169.254.170.2/32
+# This address will only exist on systems that have already set up the routes
+$ip = (Get-NetIPAddress -InterfaceIndex $ifIndex -IPAddress 169.254.170.2 -PrefixLength 32)
+if(!($ip)) {
```

Comment:
As a windows noob, I'd really appreciate it if you could add a comment for each line here.